flytekit/common/component_nodes.py

```diff
@@ -0,0 +1,140 @@
+        _logging.debug("Fetching task template for {} from Admin".format(base_model.reference_id))
+        project = base_model.reference_id.project
+        domain = base_model.reference_id.domain
+        name = base_model.reference_id.name
+        version = base_model.reference_id.version
+        sdk_task = _task.SdkTask.fetch(project, domain, name, version)
+        return cls(sdk_task)
+
+
+class SdkWorkflowNode(_six.with_metaclass(_sdk_bases.ExtendedSdkType, _workflow_model.WorkflowNode)):
+    def __init__(self, sdk_workflow=None, sdk_launch_plan=None):
+        """
+        :param flytekit.common.workflow.SdkWorkflow sdk_workflow:
+        :param flytekit.common.launch_plan.SdkLaunchPlan sdk_launch_plan:
+        """
+        self._sdk_workflow = sdk_workflow
+        self._sdk_launch_plan = sdk_launch_plan
+        super(SdkWorkflowNode, self).__init__()
+
+    @property
+    def launchplan_ref(self):
+        """
+        [Optional] A globally unique identifier for the launch plan.  Should map to Admin.
+        :rtype: flytekit.models.core.identifier.Identifier
+        """
+        return self._sdk_launch_plan.id if self._sdk_launch_plan else None
+
+    @property
+    def sub_workflow_ref(self):
+        """
+        [Optional] Reference to a subworkflow, that should be defined with the compiler context.
+        :rtype: flytekit.models.core.identifier.Identifier
+        """
+        return self._sdk_workflow.id if self._sdk_workflow else None
+
+    @property
+    def sdk_launch_plan(self):
+        """
+        :rtype: flytekit.common.launch_plan.SdkLaunchPlan
+        """
+        return self._sdk_launch_plan
+
+    @property
+    def sdk_workflow(self):
+        """
+        :rtype: flytekit.common.workflow.SdkWorkflow
+        """
+        return self._sdk_workflow
+
+    @classmethod
+    def promote_from_model(cls, base_model, sub_workflows=None, tasks=None):
+        """
+        :param flytekit.models.core.workflow.WorkflowNode base_model:
+        :param list[flytekit.models.core.workflow.WorkflowTemplate] sub_workflows:
+        :param list[flytekit.models.task.TaskTemplate] tasks:
+        :rtype: SdkWorkflowNode
+        """
+        # put the import statement here to prevent circular dependency error
+        from flytekit.common import workflow as _workflow, launch_plan as _launch_plan
+
+        project = base_model.reference.project
+        domain = base_model.reference.domain
+        name = base_model.reference.name
+        version = base_model.reference.version
+        if base_model.launchplan_ref is not None:
+            sdk_launch_plan = _launch_plan.SdkLaunchPlan.fetch(project, domain, name, version)
+            return cls(sdk_launch_plan=sdk_launch_plan)
+        elif base_model.sub_workflow_ref is not None:
+            sub_workflows = sub_workflows or []
+            # The workflow templates for sub-workflows should have been included in the original response
+            for sw in sub_workflows:
```

Comment:
nit: same comment as above about caller making this a dict.

Reply:
@matthewphsmith look through this commit? https://github.com/lyft/flytekit/pull/75/commits/cb8b07790397713b31e7aa6ef11992da1d8d5c6f